src/Cart/Mixin.js

```diff
@@ -21,11 +21,15 @@ export default {
         cartDiscountPercentage() {
             if (!this.cartCollection.discount.percentage) return 0.00
 
-            if (this.cartCollection.discount.percentage === '100.00') {
-                this.currentCheckout.payment = { provider: 'free' }
-            } else {
-                this.currentCheckout.payment = { provider: '' }
```

Comment:
Why don't we just remove this `else` statement to resolve the issue, instead of rewriting the entire block? I think we added this for situations where someone applied a 100% and then removed it. Which in reality will never happen right?!

Reply:
Oh and please could you also document any changes in the changelog?

Reply:
If you're ok with me just removing the `else` statement then I'm happy to do so. Didn't do this as we specifically added it in a different PR but as you say, it's covering us for a reasonably unrealistic situation.

Reply:
Yeah I think we should just remove it. Your new solution doesn't address that situation either. If you applied a 100% discount, continue to next, then go back, change the discount, the system won't know what payment provider you had originally selected either.